app/src/actions/vesselInfo.js

```diff
@@ -0,0 +1,129 @@
+import { SET_VESSEL_DETAILS, SET_VESSEL_TRACK } from '../constants';
+import _ from 'lodash';
+import PelagosClient from '../lib/pelagosClient';
+
+export function setCurrentVessel(vesselDetails) {
+  return (dispatch, getState) => {
+    if (!vesselDetails) {
+      dispatch({
+        type: SET_VESSEL_DETAILS,
+        payload: {}
+      });
+      return;
+    }
+    const state = getState();
+    const token = state.user.token;
+    const seriesGroup = vesselDetails.seriesgroup;
+    let request;
+
+    if (typeof XMLHttpRequest !== 'undefined') {
```

Comment:
the bigger question is: fetch vs vanilla. @clementprdhomme @nerik thought?
once we have an agreement, we need to refactor all API calls